miasm2/jitter/codegen.py

```diff
@@ -244,24 +253,51 @@ def gen_c_assignments(self, assignblk):
                     # Dont mask float affectation
                     c_main.append(
                         '%s = (%s);' % (self.id_to_c(new_dst), self.id_to_c(src)))
-                else:
+                elif new_dst.size <= self.translator.NATIVE_INT_MAX_SIZE:
                     c_main.append(
                         '%s = (%s)&%s;' % (self.id_to_c(new_dst),
                                            self.id_to_c(src),
                                            SIZE_TO_MASK[src.size]))
+                else:
+                    c_main.append(
+                        '%s = bignum_mask(%s, %d);' % (
+                            self.id_to_c(new_dst),
+                            self.id_to_c(src),
+                            src.size
+                        )
+                    )
             elif isinstance(dst, ExprMem):
                 ptr = dst.arg.replace_expr(prefetchers)
-                new_dst = ExprMem(ptr, dst.size)
-                str_dst = self.id_to_c(new_dst).replace('MEM_LOOKUP', 'MEM_WRITE')
-                c_mem.append('%s, %s);' % (str_dst[:-1], self.id_to_c(src)))
+                if ptr.size <= self.translator.NATIVE_INT_MAX_SIZE:
+                    new_dst = ExprMem(ptr, dst.size)
+                    str_dst = self.id_to_c(new_dst).replace('MEM_LOOKUP', 'MEM_WRITE')
+                    c_mem.append('%s, %s);' % (str_dst[:-1], self.id_to_c(src)))
+                else:
+                    ptr_str = self.id_to_c(ptr)
+                    if ptr.size <= self.translator.NATIVE_INT_MAX_SIZE:
+                        c_mem.append('%s, %s);' % (str_dst[:-1], self.id_to_c(src)))
+                    else:
+                        if src.size <= self.translator.NATIVE_INT_MAX_SIZE:
+                            c_mem.append('MEM_WRITE_BN_INT(jitcpu, %d, %s, %s);' % (
+                                src.size, ptr_str, self.id_to_c(src))
+                            )
+                        else:
+                            c_mem.append('MEM_WRITE_BN_BN(jitcpu, %d, %s, %s);' % (
+                                src.size, ptr_str, self.id_to_c(src))
+                            )
             else:
                 raise ValueError("Unknown dst")
 
         for dst, new_dst in dst_var.iteritems():
             if dst == self.ir_arch.IRDst:
                 continue
-            c_updt.append('%s = %s;' % (self.id_to_c(dst), self.id_to_c(new_dst)))
-            c_var.append("uint%d_t %s;" % (new_dst.size, new_dst))
+
+            if dst.size <= self.translator.NATIVE_INT_MAX_SIZE:
+                c_updt.append('%s = %s;' % (self.id_to_c(dst), self.id_to_c(new_dst)))
```

Comment:
This line can be extracted from the if/else

Reply:
Done